src/cdk/drag-drop/drop-list-ref.ts

```diff
@@ -695,6 +703,20 @@ export class DropListRef<T = any> {
     return itemOffset;
   }
 
+  /**
+   * Checks if pointer is entering in the first position
+   * @param pointerX Position of the user's pointer along the X axis.
+   * @param pointerY Position of the user's pointer along the Y axis.
+   */
+  private _isEnteringFromStart(pointerX: number, pointerY: number) {
```

Comment:
Should this account for RTL layouts?

Reply:
Good point, I've refactored the code to handle reversed containers.
Also added unit tests. Thanks